WordPress/playground-tools
# WP-NOW: Simplify multipart requests by dropping manually encoding

## What?



Drop manually encoding multipart requests. This PR is a follow-up of https://github.com/WordPress/playground-tools/pull/203.

## Why?



This PR follows [this discussion](https://github.com/WordPress/playground-tools/pull/203#discussion_r1598539790) regarding the necessity of manually encoding multipart requests. Additionally, it addresses https://github.com/WordPress/playground-tools/issues/259.

## How?



- Remove Express `fileUpload` middleware to avoid modifying the request object.
- Remove `encodeAsMultipart` helper, since we don't need to process them differently ([Playground CLI example](https://github.com/WordPress/wordpress-playground/blob/946dd7225cec1b5bebbe33072b246609d9e4524a/packages/playground/cli/src/server.ts#L28-L41)).

## Testing Instructions






1. Run the command: `nvm use && npm install`.
2. Run the command `nx preview wp-now start`.
3. Open the WP-admin page (e.g. `http://localhost:8881/wp-admin`).
4. Navigate to the Media page.
5. Upload a media file.
6. Observe the media has been uploaded correctly to the Media Library.
